python/ray/experimental/async_plasma.py

```diff
@@ -36,19 +33,30 @@ def close(self):
                 fut.cancel()
 
     def _complete_future(self, futures: List[asyncio.Future], ray_object_id):
+        logger.debug(
+            "Completing plasma futures for object id {}".format(ray_object_id))
+
         obj = ray.get(ray_object_id)
         for fut in futures:
-            try:
-                fut.set_result(obj)
-            except asyncio.InvalidStateError:
-                # Avoid issues where process_notifications
-                # and check_ready both get executed
-                pass
-        try:
```

Comment:
this is the only place where we pop from waiting_dict now, so no need to error check